test/module/irohad/multi_sig_transactions/state_test.cpp

```diff
@@ -25,228 +13,338 @@ using namespace std;
+        batch->transactions().begin(),
+        batch->transactions().end(),
+        [&time](const auto &tx) { return tx->createdTime() < time; });
   }
 };
 
+/**
+ * @given state with one transaction
+ * @when  call erase by time, where batch will be already expired
+ * @then  checks that expired state contains batch and initial doesn't conitain
+ */
 TEST(StateTest, TimeIndexInsertionByTx) {
-  log_->info("Insert one transaction with many signatures => erase tx by time");
-
-  auto quorum = 3u;
+  auto quorum = 2u;
   auto time = iroha::time::now();
 
   auto state = MstState::empty(std::make_shared<TimeTestCompleter>());
 
-  state += makeTx(1, time, makeKey(), quorum);
-  state += makeTx(1, time, makeKey(), quorum);
-
-  ASSERT_EQ(1, state.getTransactions().size());
+  state += addSignatures(makeTestBatch(txBuilder(1, time, quorum)),
+                         0,
+                         makeSignature("1_1", "1_1"));
 
   auto expired_state = state.eraseByTime(time + 1);
-  ASSERT_EQ(1, expired_state.getTransactions().size());
-  ASSERT_EQ(0, state.getTransactions().size());
+  ASSERT_EQ(1, expired_state.getBatches().size());
+  ASSERT_EQ(0, state.getBatches().size());
 }
 
+/**
+ * @given init two states
+ * @when  merge them
+ * AND make states expired
+ * @then checks that all expired transactions are preserved in expired state
+ */
 TEST(StateTest, TimeIndexInsertionByAddState) {
-  log_->info("Fill two states => add one to another => erase tx by time");
-
   auto quorum = 3u;
   auto time = iroha::time::now();
 
   auto state1 = MstState::empty(std::make_shared<TimeTestCompleter>());
-  state1 += makeTx(1, time, makeKey(), quorum);
-  state1 += makeTx(1, time, makeKey(), quorum);
+  state1 += addSignatures(makeTestBatch(txBuilder(1, time, quorum)),
+                          0,
+                          makeSignature("1_1", "1_1"));
+  state1 += addSignatures(makeTestBatch(txBuilder(1, time, quorum)),
+                          0,
+                          makeSignature("1_2", "1_2"));
 
   auto state2 = MstState::empty(std::make_shared<TimeTestCompleter>());
-  state2 += makeTx(5, time, makeKey(), quorum);
-  state2 += makeTx(6, time, makeKey(), quorum);
+  state2 += addSignatures(
+      makeTestBatch(txBuilder(2, time)), 0, makeSignature("2", "2"));
+  state2 += addSignatures(
+      makeTestBatch(txBuilder(3, time)), 0, makeSignature("3", "3"));
 
   auto completed_state = state1 += state2;
-  ASSERT_EQ(0, completed_state.getTransactions().size());
+  ASSERT_EQ(0, completed_state.getBatches().size());
 
   auto expired_state = state1.eraseByTime(time + 1);
-  ASSERT_EQ(3, expired_state.getTransactions().size());
-  ASSERT_EQ(0, state1.getTransactions().size());
-  ASSERT_EQ(2, state2.getTransactions().size());
+  ASSERT_EQ(3, expired_state.getBatches().size());
+  ASSERT_EQ(0, state1.getBatches().size());
+  ASSERT_EQ(2, state2.getBatches().size());
 }
 
+/**
+ * @given first state with two batches, seconds is empty
```

Comment:
-> given a state with two expired batches AND an empty state
when the second state is removed from the first state
AND erase by time is called
then the resulting state does not contain any transactions